source/common/stats/stats_impl.cc

```diff
@@ -355,16 +333,25 @@ void HeapRawStatDataAllocator::free(RawStatData& data) {
 
 void RawStatData::initialize(absl::string_view key) {
   ASSERT(!initialized());
-  if (key.size() > Stats::RawStatData::maxNameLength()) {
+  ref_count_ = 1;
+
+  uint64_t xfer_size = key.size();
+  memcpy(name_, key.data(), xfer_size);
```

Comment:
so where is initialize called now? Does this need some safety guard for cope size? If not, can you add a comment?

Reply:
So there are two initialize fns now -- one, `RawStatData::initialize(key)`, which is used in cases where no truncation is performed (like in `HeapRawStatDataAllocator::alloc(name)`). The other, `RawStatData::truncateAndInit(key, stats_options)`, gets called in `BlockMemoryHashSet::insert(key)`, which has access to a local `stats_options_` struct. So the former case doesn't guard for some max size.

Reply:
Can we somehow assert that we aren't going to overwrite memory here? It's not super clear to me how we know that this is a safe copy (or if heap allocated pass in the size of the heap allocation and assert that)?